hangman.py

```diff
@@ -0,0 +1,308 @@
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+
+    guesses_remaining = 6
+    warnings_remaining = 3
+    letters_guessed = []
+    print('''Welcome to the game Hangman!
+I am thinking of a word that is {} letters long.
+You have {} warnings left'''.format(len(secret_word), warnings_remaining))
+    while (guesses_remaining > 0):
+        print('''-------------
+You have {} guesses left.
+Available letters: {}'''.format(guesses_remaining, get_available_letters(letters_guessed)))
+
+        letters_guessed.append((input("Please guess a letter:")).lower())
+
+        if secret_word == get_guessed_word(secret_word, letters_guessed):
+            print("Congratulations, you won! Your total score for this game is:", guesses_remaining * len(secret_word))
+            exit()
+
+        if not letters_guessed[-1].isalpha():
+            if warnings_remaining == 0:
+                guesses_remaining -= 1
+            else:
+                warnings_remaining -= 1
+            print("Oops! That is not a valid letter. You have", warnings_remaining, "warnings left:", end="")
+        elif letters_guessed[-1] in letters_guessed[0:-1]:
+            if warnings_remaining == 0:
+                guesses_remaining -= 1
+            else:
+                warnings_remaining -= 1
+            print("Oops! You've already guessed that letter. You have", warnings_remaining, "warnings left:", end="")
+
+        elif not (letters_guessed[-1] in secret_word):
+            if letters_guessed[-1] in "aeiou":
+                guesses_remaining -= 2
+            else:
+                guesses_remaining -= 1
+            print("Oops! That letter is not in my word:", end="")
+
+        elif letters_guessed[-1] in secret_word:
+            print("Good guess:", end="")
+
+        print(get_guessed_word(secret_word, letters_guessed))
+    print("Sorry, you ran out of guesses. The word was else\nIt's", secret_word)
+
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+# (hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    my_word = ''.join(my_word.split())
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if my_word[i] != ("_"):
+                if my_word[i] != other_word[i]:
+                    return False
+                    break
+        return True
+    else:
+        return False
```

Comment:
You can just write `if len(my_word) != len(other_word):` to return False, and no `else` will be needed

Reply:
If I write your condition it will be another one after main block of code where works loop "for", is it justified?
```
    if len(my_word) == len(other_word):
        for i in range(len(my_word)):
            if my_word[i] != ("_"):
                if my_word[i] != other_word[i]:
                    return False
        return True
    else:
        return False
```

Reply:
That's what I mean:
```
    if len(my_word) != len(other_word):
        return False

    for i in range(len(my_word)):
        if my_word[i] != ("_"):
            if my_word[i] != other_word[i]:
                return False
    return True
```